examples/web_agent/worker.py

```diff
@@ -0,0 +1,203 @@
+
+        if not session.data.cdp_ws_url:
+            raise ValueError("Unable to get cdp url")
+
+        session_id = session.data.id if session.data else None
+        print("Created airtop session", session_id)
+
+        # Create a new window and navigate to the URL
+        window = client.windows.create(session_id, url=LOGIN_URL)
+        window_info = client.windows.get_window_info(session_id, window.data.window_id)
+
+        # Check whether the user is logged in
+        print("Determining whether the user is logged in...")
+        logged_in_schema = IS_LOGGED_IN_OUTPUT_SCHEMA
+        is_logged_in_response = client.windows.page_query(
+            session_id,
+            window.data.window_id,
+            prompt=IS_LOGGED_IN_PROMPT,
+            configuration=PageQueryConfig(output_schema=logged_in_schema),
+        )
+        print("Response if user is logged in")
+        parsed_response = json.loads(is_logged_in_response.data.model_response)
+        if "error" in parsed_response:
+            raise ValueError(parsed_response.error)
+        is_user_logged_in = parsed_response["isLoggedIn"]
+
+        # Prompt the user to log in if not already logged in
+        if not is_user_logged_in:
+            print(
+                f"Log into your Glassdoor account. Press Enter once done. Live view URL: {window_info.data.live_view_url}"
+            )
+            input()
+            print(
+                f"To avoid logging in again, use the profileId next time: {session.data.profile_id}"
+            )
+        else:
+            print(
+                f"User is already logged in. Live view URL: {window_info.data.live_view_url}"
+            )
+
+        # Navigate to the target URL
+        print("Navigating to target URL...")
+        client.windows.load_url(session_id, window.data.window_id, url=TARGET_URL)
+        print("Prompting the AI agent, waiting for a response...")
+
+        extract_data_response = client.windows.page_query(
+            session_id,
+            window.data.window_id,
+            prompt=EXTRACT_DATA_PROMPT,
+            configuration=PageQueryConfig(output_schema=EXTRACT_DATA_OUTPUT_SCHEMA),
+        )
+        formatted_json = json.dumps(
+            json.loads(extract_data_response.data.model_response), indent=2
+        )
+
+        # check if db exists, if not then it'll be created
+        try:
+            print("Existing workers")
+            print(dataframe_from_query_dev("SELECT * from workers limit 5"))
+        except:
+            for statement in sql_statements:
+                insert_tables(statement)
+
+        workers = find_workers(TARGET_URL, EXTRACT_DATA_PROMPT)
+        if not workers:
+            create_worker([(TARGET_URL, EXTRACT_DATA_PROMPT)])
+        print("Response:\n\n", formatted_json)
+
+        old_content = retrieve_previous_result(TARGET_URL, EXTRACT_DATA_PROMPT)
+        if old_content:
+            promptContentResponse = client.windows.page_query(
```

Comment:
Add a print statement here inside the if to let the user know that the comparison is being executed and that it might take a few mins. 

I thought the script was hanging after I got a response and the session didn't terminate. I then tried to rerun it and it failed since the profile wasn't ready. This was my user error but having that print statement would have made it clear that I needed to wait.